documentation/readme-cSharp.md

```diff
@@ -0,0 +1,344 @@
+  Expected: True
+  Actual:   False
+  ```
+
+* If the property returns task it is considered a property and should throw an exception if the test fails
+
+    ```CSharp
+    internal static Task Fast()
+    {
+      return Task.CompletedTask;
+    }
+
+    [Property]
+    public async Task Async_Task_property(
+      int i)
+    {
+      await Fast();
+      Assert.True(i == i);
+    }
+
+    [Property]
+    public Task Task_property(
+      int i)
+    {
+      Assert.True(i == i);
+      return Task.CompletedTask;
+    }
+    ```
+
+### `Property` Configuration
+---
+The `Property` attribute's constructor may take several arguments:   
+* [`AutoGenConfig`](#autogenconfig): Allow the manual control of generators
+* [`Tests`](#tests-count): Specifies the number of tests to run 
+* [`Shrinks`](#shrinks-count): Specifies the number of shrinks
+
+The `Property` attribute extends `Xunit.FactAttribute`,  so it may also take `DisplayName`, `Skip`, and `Timeout`.
+
+#### `AutoGenConfig`
+---
+The default generator used to create arguments is `GenX.defaults`. To specify different generators:
+* Create a class with a single static property or method that returns an instance of `AutoGenConfig`. 
+* Provide the type of this class as an argument to the `Property` attribute. This works around the constraint that [`Attribute` parameters must be a constant.](https://stackoverflow.com/a/33007272)
+
+  ```CSharp
+  public class AutoGenConfigContainer
+  {
+    public static AutoGenConfig _ =>
+      GenX.defaults.WithGenerator(Gen.Int32(Range.FromValue(13)));
+  }
+
+  [Property(typeof(AutoGenConfigContainer))]
+  public bool This_test_passes_because_always_13(
+    int i) =>
+    i == 13;
+  ```
+
+#### `Tests` (count)
+---
+Specifies the number of tests to be run, though more or less may occur due to shrinking or early failure.
+
+```CSharp
+[Property(tests: 3)]
+public void This_runs_3_times() => _output.WriteLine($"Test run");
+```
+
+##### `Size`
+---
+Sets the `Size` to a value for all runs.
+
+```f#
+[<Property(Size = 2)>]
+let ``"i" mostly ranges between -1 and 1`` i =
+  printfn "%i" i
+```
+
+#### Properties attribute
+---
+This optional attribute can decorate modules or classes. It sets default arguments for [`AutoGenConfig`](#autogenconfig), [`Tests`](#tests-count), [`Shrinks`](#shrinks-count), and [`Size`](#size). These will be overridden by any arguments provided by the `Property` attribute.
+
+```CSharp
+public class Int13
+{
+  public static AutoGenConfig _ => GenX.defaults.WithGenerator(Hedgehog.Gen.constant(13));
+}
+
+public class Int2718
+{
+  public static AutoGenConfig _ => GenX.defaults.WithGenerator(Hedgehog.Gen.constant(2718));
+}
+
+[Properties(typeof(Int13),1)]
+public class PropertiesSample
+{
+  [Property]
+  public void this_passes_and_runs_1(
+    int i) =>
+    i = 13;
+
+  [Property(typeof(Int2718), 2)]
+  public void this_passes_passes_and_runs_twice(
+    int i) =>
+    i = 2718;
```

Comment:
This is an assignment...?

Reply:
Thank you; no, it's because I copied the F# sample and did not convert equality, updated.